ddl/ddl_tiflash_test.go

```diff
@@ -173,17 +181,21 @@ func TempDisableEmulatorGC() func() {
 	return f
 }
 
-func (s *tiflashDDLTestSuite) SetPdLoop(tick uint64) func() {
+func (s *tiflashContext) SetPdLoop(tick uint64) func() {
 	originValue := ddl.PullTiFlashPdTick.Swap(tick)
 	return func() {
 		ddl.PullTiFlashPdTick.Store(originValue)
 	}
 }
 
 // Run all kinds of DDLs, and will create no redundant pd rules for TiFlash.
-func (s *tiflashDDLTestSuite) TestTiFlashNoRedundantPDRules(c *C) {
+func TestTiFlashNoRedundantPDRules(t *testing.T) {
+	s := createTiFlashContext(t)
+	defer teardownHelper(s)
```

Comment:
I think you could return `tiflashcontext, testkit, cleanfn := createTiFlashContext(t); defer cleanfn()`. Some other testify migration are doing so.